default/python/AI/turn_state/__init__.py

```diff
@@ -0,0 +1,39 @@
+"""
+This module contains tools to work with pre-aggregated information about AI state.
+This information is not stored in the save game and recalculated each turn.
+
+The main purpose of this module is to separate code that aggregates information and code that makes a decision.
+Decision-making code is smart and this is the dumb module,
+that means that we try to make this module API as comfortable as possible.
+
+Module API:
+Module API is represented as several functions.
+All public functions just lazy objects.
+They initialized with the first request and cache its value till the end of the turn.
+
+Avoiding classes that are exposed to outer scope requires some additional calls inside functions,
+ but this price is not too high to get a simple API.
+
+The module is split into submodules by the idea that each submodule works with the same set of data.
```

Comment:
The module is split into sub-modules, based around the idea that each submodule should work with the same set of data.